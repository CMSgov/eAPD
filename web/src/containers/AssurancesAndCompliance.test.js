import { shallow } from 'enzyme';
import React from 'react';

import {
  plain as AssurancesAndCompliance,
  mapStateToProps,
  mapDispatchToProps,
  LinkOrText
} from './AssurancesAndCompliance';

import {
  setComplyingWithProcurement,
  setComplyingWithRecordsAccess,
  setComplyingWithSecurity,
  setComplyingWithSoftwareRights,
  setJustificationForProcurement,
  setJustificationForRecordsAccess,
  setJustificationForSecurity,
  setJustificationForSoftwareRights
} from '../actions/editApd';

describe('assurances and compliance component', () => {
  describe('main component', () => {
    const props = {
      citations: {
        procurement: [
          { title: '42 CFR Part 495.348', checked: true, explanation: '' },
          { title: 'SMM Section 11267', checked: true, explanation: '' },
          { title: '45 CFR 95.613', checked: '', explanation: '' },
          { title: '45 CFR Part 75.326', checked: false, explanation: '' }
        ],
        recordsAccess: [
          {
            title: '42 CFR Part 495.350',
            checked: false,
            explanation: 'some words'
          },
          { title: '42 CFR Part 495.352', checked: true, explanation: '' },
          { title: '42 CFR Part 495.346', checked: true, explanation: '' },
          {
            title: '42 CFR 433.112(b)',
            checked: true,
            explanation: ''
          },
          {
            title: '45 CFR Part 95.615',
            checked: true,
            explanation: 'other words'
          },
          { title: 'SMM Section 11267', checked: true, explanation: '' }
        ],
        softwareRights: [
          { title: '42 CFR Part 495.360', checked: true, explanation: '' },
          { title: '45 CFR Part 95.617', checked: true, explanation: '' },
          { title: '42 CFR Part 431.300', checked: false, explanation: '' },
          { title: '42 CFR Part 433.112', checked: true, explanation: '' }
        ],
        security: [
          {
            title: '45 CFR 164 Security and Privacy',
            checked: false,
            explanation: ''
          }
        ]
      },
      complyingWithProcurement: jest.fn(),
      complyingWithRecordsAccess: jest.fn(),
      complyingWithSecurity: jest.fn(),
      complyingWithSoftwareRights: jest.fn(),
      justificationForProcurement: jest.fn(),
      justificationForRecordsAccess: jest.fn(),
      justificationForSecurity: jest.fn(),
      justificationForSoftwareRights: jest.fn()
    };

    beforeEach(() => {
      props.complyingWithProcurement.mockClear();
      props.complyingWithRecordsAccess.mockClear();
      props.complyingWithSecurity.mockClear();
      props.complyingWithSoftwareRights.mockClear();
      props.justificationForProcurement.mockClear();
      props.justificationForRecordsAccess.mockClear();
      props.justificationForSecurity.mockClear();
      props.justificationForSoftwareRights.mockClear();
    });

    test('renders correctly', () => {
      expect(shallow(<AssurancesAndCompliance {...props} />)).toMatchSnapshot();
    });

    test('dispatches when a citation is toggled yes/no', () => {
      const component = shallow(<AssurancesAndCompliance {...props} />);

      // Based on the state above, there are four sections to test. We need
      // to target one radio button in each section. There are two per
      // assurance item, so the indices we need to target are:
      //   Procurement: 0/1 (yes/no)
      //   Records access: 8/9
      //   Software rights: 20/21
      //   Security: 28/29

      component
        .find('ChoiceComponent')
        .at(0)
        .simulate('change');
      component
        .find('ChoiceComponent')
        .at(9)
        .simulate('change');
      component
        .find('ChoiceComponent')
        .at(21)
        .simulate('change');
      component
        .find('ChoiceComponent')
        .at(28)
        .simulate('change');

      expect(props.complyingWithProcurement).toHaveBeenCalledWith(0, true);
      expect(props.complyingWithRecordsAccess).toHaveBeenCalledWith(0, false);
      expect(props.complyingWithSoftwareRights).toHaveBeenCalledWith(0, false);
      expect(props.complyingWithSecurity).toHaveBeenCalledWith(0, true);
    });

    test('dispatches when text is changed', () => {
      const component = shallow(<AssurancesAndCompliance {...props} />);

      // Choice component indices from above:
      //   Procurement: 0/1 (yes/no)
      //   Records access: 8/9
      //   Software rights: 20/21
      //   Security: 28/29
      // In this case, we only want the "no" choices, and we'll render the
      // "checkedChildren" prop to access the actual TextArea. Then we'll
      // simulate changing that.

      shallow(
        component
          .find('ChoiceComponent')
          .at(1)

          .prop('checkedChildren')
      )
        .find('TextArea')
        .simulate('change', { target: { value: 'new text 1' } });

      shallow(
        component
          .find('ChoiceComponent')
          .at(9)

          .prop('checkedChildren')
      )
        .find('TextArea')
        .simulate('change', { target: { value: 'new text 2' } });

      shallow(
        component
          .find('ChoiceComponent')
          .at(21)

          .prop('checkedChildren')
      )
        .find('TextArea')
        .simulate('change', { target: { value: 'new text 3' } });

      shallow(
        component
          .find('ChoiceComponent')
          .at(29)

          .prop('checkedChildren')
      )
        .find('TextArea')
        .simulate('change', { target: { value: 'new text 4' } });

      expect(props.justificationForProcurement).toHaveBeenCalledWith(
        0,
        'new text 1'
      );
      expect(props.justificationForRecordsAccess).toHaveBeenCalledWith(
        0,
        'new text 2'
      );
      expect(props.justificationForSoftwareRights).toHaveBeenCalledWith(
        0,
        'new text 3'
      );
      expect(props.justificationForSecurity).toHaveBeenCalledWith(
        0,
        'new text 4'
      );
    });
  });

  it('maps state to props', () => {
    expect(
      mapStateToProps({
        apd: {
          data: {
            federalCitations: 'this gets mapped over'
          }
        }
      })
    ).toEqual({ citations: 'this gets mapped over' });
  });

  it('maps dispatch to props', () => {
    expect(mapDispatchToProps).toEqual({
      complyingWithProcurement: setComplyingWithProcurement,
      complyingWithRecordsAccess: setComplyingWithRecordsAccess,
      complyingWithSecurity: setComplyingWithSecurity,
      complyingWithSoftwareRights: setComplyingWithSoftwareRights,
      justificationForProcurement: setJustificationForProcurement,
      justificationForRecordsAccess: setJustificationForRecordsAccess,
      justificationForSecurity: setJustificationForSecurity,
      justificationForSoftwareRights: setJustificationForSoftwareRights
    });
  });

  it('LinkOrText component renders correctly', () => {
    expect(shallow(<LinkOrText title="hello" />)).toMatchSnapshot();
    expect(shallow(<LinkOrText title="hello" link="url" />)).toMatchSnapshot();
  });
});
