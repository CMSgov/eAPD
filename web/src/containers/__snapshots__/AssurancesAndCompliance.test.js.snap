// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assurances and compliance component LinkOrText component renders correctly 1`] = `"hello"`;

exports[`assurances and compliance component LinkOrText component renders correctly 2`] = `
<a
  href="url"
  rel="noopener noreferrer"
  target="_blank"
>
  hello
</a>
`;

exports[`assurances and compliance component main component renders correctly 1`] = `
<Section
  id="assurances-compliance"
  resource="assurancesAndCompliance"
>
  <Subsection
    headerClassName={null}
    id="assurances-compliance-fed-citations"
    nested={false}
    resource="assurancesAndCompliance.citations"
  >
    <div
      className="ds-u-margin-bottom--3"
      key="procurement"
    >
      <h4
        className="ds-h4"
      >
        Procurement Standards (Competition / Sole Source)
      </h4>
      <fieldset
        className="ds-u-margin-top--2"
        key="42 CFR Part 495.348"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=60f02a079c6f091871999372387e5750&mc=true&node=se42.5.495_1348&rgn=div8"
              title="42 CFR Part 495.348"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-procurement-42_CFR_Part_495.348"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-procurement-42_CFR_Part_495.348"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-procurement-42_CFR_Part_495.348"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="SMM Section 11267"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.cms.gov/Regulations-and-Guidance/Guidance/Manuals/Paper-Based-Manuals-Items/CMS021927.html"
              title="SMM Section 11267"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-procurement-SMM_Section_11267"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-procurement-SMM_Section_11267"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-procurement-SMM_Section_11267"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="45 CFR 95.613"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/retrieveECFR?gp=1&SID=6d93b829c0db163a0124bfeae1704e42&ty=HTML&h=L&mc=true&r=SECTION&n=se45.1.95_1613"
              title="45 CFR 95.613"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={false}
          label="Yes"
          name="apd-assurances-yes-explanation-procurement-45_CFR_95.613"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-procurement-45_CFR_95.613"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-procurement-45_CFR_95.613"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="45 CFR Part 75.326"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link={null}
              title="45 CFR Part 75.326"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={false}
          label="Yes"
          name="apd-assurances-yes-explanation-procurement-45_CFR_Part_75.326"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={true}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-procurement-45_CFR_Part_75.326"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-procurement-45_CFR_Part_75.326"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
    </div>
    <div
      className="ds-u-margin-bottom--3"
      key="recordsAccess"
    >
      <h4
        className="ds-h4"
      >
        Access to Records, Reporting and Agency Attestations
      </h4>
      <fieldset
        className="ds-u-margin-top--2"
        key="42 CFR Part 495.350"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=8c677ad13ab6c202c5ae1ee1c1c0b487&mc=true&node=se42.5.495_1350&rgn=div8"
              title="42 CFR Part 495.350"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={false}
          label="Yes"
          name="apd-assurances-yes-explanation-recordsAccess-42_CFR_Part_495.350"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={true}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-recordsAccess-42_CFR_Part_495.350"
                onChange={[Function]}
                rows={5}
                value="some words"
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-recordsAccess-42_CFR_Part_495.350"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="42 CFR Part 495.352"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=8c677ad13ab6c202c5ae1ee1c1c0b487&mc=true&node=se42.5.495_1352&rgn=div8"
              title="42 CFR Part 495.352"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-recordsAccess-42_CFR_Part_495.352"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-recordsAccess-42_CFR_Part_495.352"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-recordsAccess-42_CFR_Part_495.352"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="42 CFR Part 495.346"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=8c677ad13ab6c202c5ae1ee1c1c0b487&mc=true&node=se42.5.495_1346&rgn=div8"
              title="42 CFR Part 495.346"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-recordsAccess-42_CFR_Part_495.346"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-recordsAccess-42_CFR_Part_495.346"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-recordsAccess-42_CFR_Part_495.346"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="42 CFR 433.112(b)"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=8c677ad13ab6c202c5ae1ee1c1c0b487&mc=true&node=se42.4.433_1112&rgn=div8"
              title="42 CFR 433.112(b)"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-recordsAccess-42_CFR_433.112(b)"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-recordsAccess-42_CFR_433.112(b)"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-recordsAccess-42_CFR_433.112(b)"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="45 CFR Part 95.615"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=2cfd59b6ec7c0a6adf749237ebbd1ba1&mc=true&node=se45.1.95_1615&rgn=div8"
              title="45 CFR Part 95.615"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-recordsAccess-45_CFR_Part_95.615"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-recordsAccess-45_CFR_Part_95.615"
                onChange={[Function]}
                rows={5}
                value="other words"
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-recordsAccess-45_CFR_Part_95.615"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="SMM Section 11267"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.cms.gov/Regulations-and-Guidance/Guidance/Manuals/Paper-Based-Manuals-Items/CMS021927.html"
              title="SMM Section 11267"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-recordsAccess-SMM_Section_11267"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-recordsAccess-SMM_Section_11267"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-recordsAccess-SMM_Section_11267"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
    </div>
    <div
      className="ds-u-margin-bottom--3"
      key="softwareRights"
    >
      <h4
        className="ds-h4"
      >
        Software & Ownership Rights, Federal Licenses, Information Safeguarding, HIPAA compliance, and progress reports
      </h4>
      <fieldset
        className="ds-u-margin-top--2"
        key="42 CFR Part 495.360"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link={null}
              title="42 CFR Part 495.360"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-softwareRights-42_CFR_Part_495.360"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-softwareRights-42_CFR_Part_495.360"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-softwareRights-42_CFR_Part_495.360"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="45 CFR Part 95.617"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link={null}
              title="45 CFR Part 95.617"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-softwareRights-45_CFR_Part_95.617"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-softwareRights-45_CFR_Part_95.617"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-softwareRights-45_CFR_Part_95.617"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="42 CFR Part 431.300"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=8c677ad13ab6c202c5ae1ee1c1c0b487&mc=true&node=se42.4.431_1300&rgn=div8"
              title="42 CFR Part 431.300"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={false}
          label="Yes"
          name="apd-assurances-yes-explanation-softwareRights-42_CFR_Part_431.300"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={true}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-softwareRights-42_CFR_Part_431.300"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-softwareRights-42_CFR_Part_431.300"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
      <fieldset
        className="ds-u-margin-top--2"
        key="42 CFR Part 433.112"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=8c677ad13ab6c202c5ae1ee1c1c0b487&mc=true&node=se42.4.433_1112&rgn=div8"
              title="42 CFR Part 433.112"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={true}
          label="Yes"
          name="apd-assurances-yes-explanation-softwareRights-42_CFR_Part_433.112"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={false}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-softwareRights-42_CFR_Part_433.112"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-softwareRights-42_CFR_Part_433.112"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
    </div>
    <div
      className="ds-u-margin-bottom--3"
      key="security"
    >
      <h4
        className="ds-h4"
      >
        Security and interface requirements to be employed for all state HIT systems
      </h4>
      <fieldset
        className="ds-u-margin-top--2"
        key="45 CFR 164 Security and Privacy"
      >
        <legend
          className="ds-c-label"
        >
          Are you complying with
           
          <strong>
            <LinkOrText
              link="https://www.ecfr.gov/cgi-bin/text-idx?SID=2cfd59b6ec7c0a6adf749237ebbd1ba1&mc=true&tpl=/ecfrbrowse/Title45/45cfr164_main_02.tpl"
              title="45 CFR 164 Security and Privacy"
            />
          </strong>
          ?
        </legend>
        <ChoiceComponent
          checked={false}
          label="Yes"
          name="apd-assurances-yes-explanation-security-45_CFR_164_Security_and_Privacy"
          onChange={[Function]}
          size="small"
          type="radio"
          value="yes"
        />
        <ChoiceComponent
          checked={true}
          checkedChildren={
            <div
              className="ds-c-choice__checkedChild"
            >
              <TextArea
                label="Please explain"
                name="explanation-security-45_CFR_164_Security_and_Privacy"
                onChange={[Function]}
                rows={5}
                value=""
              />
            </div>
          }
          label="No"
          name="apd-assurances-no-explanation-security-45_CFR_164_Security_and_Privacy"
          onChange={[Function]}
          size="small"
          type="radio"
          value="no"
        />
      </fieldset>
    </div>
  </Subsection>
</Section>
`;
